### PR TITLE
DHFPROD-2603: e2e UI mapping step tests.

### PR DESCRIPTION
**Pull request by timur-isangulov**
In this pr:
1. Mapping step e2e UI tests.
2. Removed sleeps in some cases.
3. Remove test entity at the end of flow tests,

### CODE REVIEW

**Comment by timur-isangulov** [comment]
The tests passed on Jenkins. Can you please approve.

File: web/e2e/page-objects/flows/editFlow.ts [hunk 1]
```diff
@@ -148,6 +148,10 @@ export class EditFlow extends AppPage {
     return await menuOption.click();
   }
 
+  get stepDialogBox() {
+    return element(by.css("mat-dialog-container"));
```
Comment (ayuwono) [inline web/e2e/page-objects/flows/editFlow.ts]: is this supposed to be element or class? look at flowDialogBox below

File: web/e2e/page-objects/flows/manageFlows.ts [hunk 2]
```diff
@@ -24,6 +23,10 @@ export class ManageFlows extends AppPage {
     return await button.click();
   }
 
+  get flowDialogBox() {
+    return element(by.css(".mat-dialog-container"));
```
Comment (ayuwono) [inline web/e2e/page-objects/flows/manageFlows.ts]: is this supposed to be element or class? look at stepDialogBox above

File: web/e2e/page-objects/dashboard/dashboard.ts [hunk 3]
```diff
@@ -21,6 +21,29 @@ export class DashboardPage extends AppPage {
     return element(by.buttonText('Clear!'));
   }
 
+  async clearAllDatabases() {
```
Comment (ayuwono) [inline web/e2e/page-objects/dashboard/dashboard.ts]: consider to remove the sleeps here?

File: web/e2e/page-objects/flows/editFlow.ts [hunk 4]
```diff
@@ -177,6 +181,7 @@ export class EditFlow extends AppPage {
       await stepsPage.setStepDescription(step.stepDesc);
       browser.sleep(2000);
       await stepsPage.clickStepCancelSave("save");
+     // await browser.wait(EC.invisibilityOf(this.stepDialogBox));
```
Comment (ayuwono) [inline web/e2e/page-objects/flows/editFlow.ts]: remove it if it's not used

File: web/e2e/page-objects/flows/manageFlows.ts [hunk 5]
```diff
@@ -318,10 +331,11 @@ export class ManageFlows extends AppPage {
         await manageFlowPage.setFlowOptions(n, "value", flow.options.n[1]);
       }
     }
+    await browser.wait(EC.elementToBeClickable(manageFlowPage.flowCancelSaveButton('save')));
     await manageFlowPage.clickFlowCancelSave("save");
-    await browser.sleep(2000);
+    await browser.wait(EC.invisibilityOf(manageFlowPage.flowDialogBox));
     await browser.wait(EC.visibilityOf(manageFlowPage.manageFlowPageHeader));
-    await browser.sleep(5000);
+    //await browser.sleep(5000);
```
Comment (ayuwono) [inline web/e2e/page-objects/flows/manageFlows.ts]: remove it when it's not used

File: web/e2e/page-objects/steps/steps.ts [hunk 6]
```diff
@@ -187,13 +187,19 @@ export class Steps extends AppPage {
     return element(by.id(`step-${option}-btn`));
   }
 
+  get stepDialogBox() {
+    return element(by.css("mat-dialog-container"));
```
Comment (ayuwono) [inline web/e2e/page-objects/steps/steps.ts]: make sure it's an element or class?

File: web/e2e/specs/flows/edit-flows.ts [hunk 7]
```diff
@@ -25,10 +25,10 @@ export default function (qaProjectDir) {
     });
 
     it('Should add a flow', async function () {
-      await browser.refresh();
-      await browser.waitForAngular();
-      //await appPage.clickFlowTab();
-      //await browser.wait(EC.visibilityOf(manageFlowPage.newFlowButton));
+      //await browser.refresh();
```
Comment (ayuwono) [inline web/e2e/specs/flows/edit-flows.ts]: remove it when it's unused

File: web/e2e/specs/flows/steps/mapping.ts [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+      await browser.wait(EC.visibilityOf(editFlowPage.runFlowHeader));
+      await editFlowPage.clickButtonRunCancel("flow");
+      await browser.wait(EC.visibilityOf(editFlowPage.finishedLatestJobStatus));
+    });
+
+    it('should create mapping step with name, description and target entity', async function () {
+      await editFlowPage.addStep(flow1, mapping);
+    });
+
+    xit('should create mapping step with name, description and query', async function () {
```
Comment (ayuwono) [inline web/e2e/specs/flows/steps/mapping.ts]: remove this empty test case

File: web/e2e/page-objects/flows/editFlow.ts (hunk 1 above)
Comment (timur-isangulov) [reply web/e2e/page-objects/flows/editFlow.ts]: Both work "<mat-dialog-container aria-modal="true" class="mat-dialog-container ng-tns-c25-10 ng-trigger ng-trigger-dialogContainer ng-star-inserted" tabindex="-1""

File: web/e2e/page-objects/flows/manageFlows.ts (hunk 2 above)
Comment (timur-isangulov) [reply web/e2e/page-objects/flows/manageFlows.ts]: Both work "<mat-dialog-container aria-modal="true" class="mat-dialog-container ng-tns-c25-10 ng-trigger ng-trigger-dialogContainer ng-star-inserted" tabindex="-1""

File: web/e2e/page-objects/steps/steps.ts (hunk 6 above)
Comment (timur-isangulov) [reply web/e2e/page-objects/steps/steps.ts]: Both work "<mat-dialog-container aria-modal="true" class="mat-dialog-container ng-tns-c25-10 ng-trigger ng-trigger-dialogContainer ng-star-inserted" tabindex="-1""

File: web/e2e/page-objects/dashboard/dashboard.ts (hunk 3 above)
Comment (timur-isangulov) [reply web/e2e/page-objects/dashboard/dashboard.ts]: Removed.

File: web/e2e/page-objects/flows/editFlow.ts (hunk 4 above)
Comment (timur-isangulov) [reply web/e2e/page-objects/flows/editFlow.ts]: Removed.

File: web/e2e/page-objects/flows/manageFlows.ts (hunk 5 above)
Comment (timur-isangulov) [reply web/e2e/page-objects/flows/manageFlows.ts]: Removed.

File: web/e2e/specs/flows/edit-flows.ts (hunk 7 above)
Comment (timur-isangulov) [reply web/e2e/specs/flows/edit-flows.ts]: Removed.

File: web/e2e/specs/flows/steps/mapping.ts (hunk 8 above)
Comment (timur-isangulov) [reply web/e2e/specs/flows/steps/mapping.ts]: Removed.

**Pull request merged**